src/openrct2/management/finance.c

```diff
@@ -76,6 +76,11 @@ void finance_payment(money32 amount, rct_expenditure_type type)
 	money32 new_money = cur_money - amount;
 
 	//overflow check
+	if (amount < 0 && cur_money >= INT_MAX + amount)
```

Comment:
It is indeed a little bit tricky to check for overflow without actually causing it.